src/main/java/org/sagebionetworks/bridge/spring/controllers/AuthenticationController.java

```diff
@@ -291,6 +292,26 @@ public StatusMessage resetPassword() throws Exception {
         authenticationService.resetPassword(passwordReset);
         return new StatusMessage("Password has been changed.");
     }
+    
+    @PostMapping("/v3/auth/oauth/signIn")
+    public JsonNode oauthSignIn() {
+        OAuthAuthorizationToken token = parseJson(OAuthAuthorizationToken.class);
+        
+        Study study = studyService.getStudy(token.getStudyId());
+        CriteriaContext context = getCriteriaContext(study.getStudyIdentifier());
+        
+        UserSession session = null;
+        try {
+            session = authenticationService.oauthSignIn(context, token);
+        } catch(ConsentRequiredException e) {
+            setCookieAndRecordMetrics(e.getUserSession());
+            throw e;
+        }
```

Comment:
Similarly, we don't need to check consent.

Reply:
Removed